bigtable/spark/src/main/scala/example/DataFrameDemo.scala

```diff
@@ -0,0 +1,92 @@
+package example
+
+import org.apache.hadoop.hbase.spark.datasources.{HBaseSparkConf, HBaseTableCatalog}
+
+import scala.util.Try
+
+object DataFrameDemo extends App {
+
+  println("Starting up...")
+
+  val projectId = args(0)
+  val instanceId = args(1)
+  val table = Try(args(2)).getOrElse("dataframe-demo")
+  val numRecords = Try(args(3).toInt).getOrElse(10)
+
+  import org.apache.spark.sql.SparkSession
+  val spark = SparkSession.builder().master("local[*]").getOrCreate()
+  println("Spark version: " + spark.version)
+
+  val catalog =
+    s"""{
+       |"table":{"namespace":"default", "name":"$table", "tableCoder":"PrimitiveType"},
+       |"rowkey":"key",
+       |"columns":{
+       |"col0":{"cf":"rowkey", "col":"key", "type":"string"},
+       |"col1":{"cf":"cf1", "col":"col1", "type":"boolean"},
+       |"col2":{"cf":"cf2", "col":"col2", "type":"double"},
+       |"col3":{"cf":"cf3", "col":"col3", "type":"int"}
+       |}
+       |}""".stripMargin
+  println(s"Writing $numRecords records to $table")
+
+  // FIXME Explain the options
+  // FIXME Where to find more options supported? Any docs?
+  val opts = Map(
+    HBaseTableCatalog.tableCatalog -> catalog,
+    HBaseTableCatalog.newTable -> "5",
+    HBaseSparkConf.USE_HBASECONTEXT -> "false") // accepts xml configs only
+
+  import spark.implicits._
+  val records = (0 until numRecords).map(BigtableRecord.apply).toDF
+
+  // TODO Use a command-line option to switch between command line params and xml
+
+  // Hack to specify HBase properties on command line
```

Comment:
After a bit more thought, I think we should get rid of hbase-site.xml altogether. Settings things like `com.google.cloud.bigtable.hbase2_x.BigtableConnection` should be considered an internal implementation detail of BigtableConfiguration. I'm in particular worried about newer versions needing other shims injected. For each if the hbase-connector decided to start using async apis then we would have to start setting the async connection class name as well. I'd rather this be maintained java-bigtable-hbase than in a customer's handwritten xml file.

Please remove the xml file altogether and only use BigtableConfiguration here